rotkehlchen/balances/historical_balances.py

```diff
@@ -0,0 +1,300 @@
+import logging
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.constants import DAY_IN_SECONDS, ZERO
+from rotkehlchen.constants.assets import A_USD
+from rotkehlchen.db.filtering import HistoryEventFilterQuery
+from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.errors.price import NoPriceForGivenTimestamp
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.types import EventDirection
+from rotkehlchen.history.price import PriceHistorian
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import Timestamp
+from rotkehlchen.utils.misc import timestamp_to_daystart_timestamp, ts_ms_to_sec
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import Asset
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
+class HistoricalBalance:
+    timestamp: Timestamp
+    amount: FVal = ZERO
+    usd_value: FVal = ZERO
+
+    def serialize(self) -> dict[str, Any]:
+        return {
+            'amount': str(self.amount),
+            'usd_value': str(self.usd_value),
+            'timestamp': self.timestamp,
+        }
+
+
+class HistoricalBalancesManager:
+    def __init__(self, database: 'DBHandler') -> None:
+        self.db = database
+
+    def get_balances(self) -> tuple[list[Timestamp], list[str]]:
+        """Retrieve historical usd balances until now, calculating cumulative sums.
+
+        When a timestamp has zero balance, we use the last non-zero value but don't
+        include it in the cumulative sum calculation.
+        """
+        if not (daily_values := self._get_daily_values(
+            query='SELECT timestamp, usd_value, usd_value FROM historical_balance_per_asset',
```

Comment:
So I see you created a new DB table. Didn't we have a discussion in the standup to not do exactly that? Please if we discuss something stick to it, or ask earlier about the approach to make sure we are in the same page.

Communicate, communicate, communicate.